### PR TITLE
Test mies nwb data

### PR DESCRIPTION
**Pull request by sgratiy**
Tests for EphysNWBData and MiesNWBData. Addressing #363

### CODE REVIEW

File: ipfx/dataset/ephys_nwb_data.py [hunk 1]
```diff
@@ -51,23 +51,17 @@ class EphysNWBData(EphysDataInterface):
...
     def __init__(self,
                  nwb_file: str,
                  ontology: StimulusOntology,
                  load_into_memory: bool = True,
-                 validate_stim: bool = True,
                  ):
 
-        super().init(ontology=ontology, validate_stim=validate_stim)
+        super().__init__(ontology=ontology)
+        load_into_memory = False # not working
```
Comment (NileGraddis) [inline ipfx/dataset/ephys_nwb_data.py]: is this a todo?

File: ipfx/dataset/ephys_data_interface.py [hunk 2]
```diff
@@ -189,15 +162,13 @@ def get_stimulus_unit(self, sweep_number: int) -> str:
         raise NotImplementedError
 
     @abc.abstractmethod
-    def get_real_sweep_number(self, sweep_name:str, assumed_sweep_number: Optional[int]=None) -> int:
+    def get_sweep_number(self, sweep_name:str) -> int:
```
Comment (NileGraddis) [inline ipfx/dataset/ephys_data_interface.py]: can this be deleted?

File: ipfx/dataset/ephys_nwb_data.py [hunk 3]
```diff
@@ -192,37 +176,30 @@ def get_session_start_time(self):
...
+        if series is None:
+            raise ValueError("No TimeSeries found for sweep number {}.".format(sweep_number))
 
-            unit = self.get_unit_name(stimulus_dataset.attrs)
-            unit_str = self.get_long_unit_name(unit)
+        stimulus = []
 
-            return unit_str
+        for s in series:
+            if isinstance(s,(CurrentClampStimulusSeries,VoltageClampStimulusSeries)):
```
Comment (NileGraddis) [inline ipfx/dataset/ephys_nwb_data.py]: is there a common base type?

File: ipfx/dataset/ephys_nwb_data.py [hunk 4]
```diff
@@ -192,37 +176,30 @@ def get_session_start_time(self):
...
-    @staticmethod
-    def get_unit_name(stim_attrs):
-        if 'unit' in stim_attrs:
-            unit = to_str(stim_attrs["unit"])
-        elif 'units' in stim_attrs:
-            unit = to_str(stim_attrs["units"])
+        if len(stimulus) == 1:
+            return stimulus[0].unit
+        elif len(stimulus)>1:
+            raise ValueError("Found multiple stimulus series for sweep number {sweep_number}")
```
Comment (NileGraddis) [inline ipfx/dataset/ephys_nwb_data.py]: would be good if the error message identified them somehow

File: ipfx/stimulus.py [hunk 5]
```diff
@@ -26,7 +26,7 @@ def has_tag(self, tag, tag_type=None):
 
 class StimulusOntology(object):
 
-    DEFAULT_STIMULUS_ONTOLOGY_FILE = os.path.join(os.path.dirname(__file__), 'defaults/stimulus_ontology.json')
+    DEFAULT_STIMULUS_ONTOLOGY_FILE = os.path.join(os.path.dirname(__file__), './defaults/stimulus_ontology.json')
```
Comment (NileGraddis) [inline ipfx/stimulus.py]: probably don't want to hardcode these forward slashes

File: ipfx/dataset/ephys_nwb_data.py (hunk 1 above)
Comment (sgratiy) [reply ipfx/dataset/ephys_nwb_data.py]: @gnayuy has fixed this on his branch

File: ipfx/dataset/ephys_data_interface.py (hunk 2 above)
Comment (sgratiy) [reply ipfx/dataset/ephys_data_interface.py]: I believe so. The logic was to try getting the "real" (explicitly assigned) sweep number. But for old nwb1 files it was missing, in which case resort to inferring sweep number from the dataset name. My understanding that for nwb2 "real" sweep number is always exists.

File: ipfx/dataset/ephys_nwb_data.py (hunk 3 above)
Comment (sgratiy) [reply ipfx/dataset/ephys_nwb_data.py]: There is (PatchClampSeries), but it is also the parent for the CurrentClampSeries, VoltageClampSeries used by the response series.

File: ipfx/stimulus.py (hunk 5 above)
Comment (NileGraddis) [reply ipfx/stimulus.py]: what about the one between defaults and stimulus ontology?

File: ipfx/dataset/ephys_data_interface.py (hunk 2 above)
Comment (NileGraddis) [reply ipfx/dataset/ephys_data_interface.py]: ok, would you mind deleting then?

Comment (sgratiy) [reply ipfx/dataset/ephys_data_interface.py]: I thought we were talking about `get_real_sweep_number`, but you want me to delete `get_sweep_number`. got it.

File: ipfx/stimulus.py (hunk 5 above)
Comment (sgratiy) [reply ipfx/stimulus.py]: I am not sure how to point to the json otherwise

**Comment by sgratiy** [comment]
@NileGraddis could you please re-review my PR. I wiped out all the sweep_map junk and added get_clamp_mode and a helper function to access series based on sweep_number `EphysNWBData._get_series`.  Since I rebased I fully re-written the history.

File: ipfx/dataset/create.py [hunk 6]
```diff
@@ -73,12 +77,16 @@ def create_ephys_data_set(nwb_file: str,
...
-            nwb_data = MIESNWB2Data(nwb_file, ontology_file)
+            labnotebook = LabNotebookReaderIgorNwb(nwb_file)
+            nwb_data = MIESNWB2Data(nwb_file, labnotebook, ontology)
+
         else:
-            nwb_data = HBGNWB2Data(nwb_file, ontology_file)
+            nwb_data = HBGNWB2Data(nwb_file, ontology)
 
     else:
         raise ValueError("Unsupported or unknown NWB major" +
```
Comment (NileGraddis) [inline ipfx/dataset/create.py]: use formatting here rather than +

File: ipfx/dataset/ephys_data_interface.py [hunk 7]
```diff
@@ -130,37 +131,6 @@ def get_stimulus_unit(self, sweep_number: int) -> str:
 
         raise NotImplementedError
 
-    @abc.abstractmethod
```
Comment (NileGraddis) [inline ipfx/dataset/ephys_data_interface.py]: nice!

File: ipfx/dataset/create.py (hunk 6 above)
Comment (NileGraddis) [reply ipfx/dataset/create.py]: (I realize this is not your change, but in the interests of good stewardship would be nice, not a requirement to merge).

File: ipfx/dataset/ephys_nwb_data.py [hunk 8]
```diff
@@ -51,30 +51,58 @@ class EphysNWBData(EphysDataInterface):
 
     """
 
-    SCALAR_ATTRS = ["bias_current",
-                    "stimulus_scale_factor",
-                    "bridge_balance",
-                    "stimulus_scale_factor"]
+    STIMULUS = (VoltageClampStimulusSeries, CurrentClampStimulusSeries)
```
Comment (NileGraddis) [inline ipfx/dataset/ephys_nwb_data.py]: good stuff

File: ipfx/dataset/ephys_nwb_data.py [hunk 9]
```diff
@@ -51,30 +51,58 @@ class EphysNWBData(EphysDataInterface):
...
     def __init__(self,
                  nwb_file: str,
                  ontology: StimulusOntology,
                  load_into_memory: bool = True,
-                 validate_stim: bool = True,
                  ):
 
-        super().init(ontology=ontology, validate_stim=validate_stim)
+        super().__init__(ontology=ontology)
+        load_into_memory = False # not working
```
Comment (NileGraddis) [inline ipfx/dataset/ephys_nwb_data.py]: If this is not working, delete it and raise a ticket.

**Pull request merged**